python/ray/data/_internal/execution/operators/hash_shuffle.py

```diff
@@ -957,6 +957,125 @@ def _estimate_aggregator_memory_allocation(
         return aggregator_total_memory_required
 
 
+@ray.remote
+class IssueDetector:
```

Comment:
Let's rebase this onto existing Issue Detection framework (actually having very similar abstractions